src/CodeGen.cpp

```diff
@@ -174,7 +174,7 @@ bool CodeGen::llvm_NVPTX_enabled = false;
 bool CodeGen::llvm_Mips_enabled = false;
 
 void CodeGen::compile(Stmt stmt, string name,
-                      const vector<Argument> &args,
+                      const ArgInfo &arg_info,
```

Comment:
FYI: I also considered splitting the args vector into two arguments ("inputs" and "outputs"), but that would have made for a lot more code churn... and also adding yet another field to Argument indicating input vs output, but that also looked like a lot of churn.